INFO:__main__:Running Tx DeliveryTxParams successful at the 2 time retry
ERROR:__main__:Errored: 1 retry happened in running tx: StockLevelTxParams, ErrorMsg: 
[ restart transaction: TransactionRetryWithProtoRefreshError: ReadWithinUncertaintyIntervalError: read at time 1635652018.919758000,0 encountered previous write with future timestamp 1635652018.941147000,0 within uncertainty interval `t <= (local=1635652019.358711000,0, global=1635652019.419758000,0)`; observed timestamps: [{1 1635652018.919758000,0} {2 1635652019.506716000,0} {4 1635652019.358711000,0} {5 1635652018.920739000,0}]: "sql txn" meta={id=12dc7238 pri=0.04130127 epo=0 ts=1635652018.919758000,0 min=1635652018.919758000,0 seq=0} lock=false stat=PENDING rts=1635652018.919758000,0 wto=false gul=1635652019.419758000,0
HINT:  See: https://www.cockroachlabs.com/docs/v21.1/transaction-retry-error-reference.html#readwithinuncertaintyinterval
 ], Traceback: 
[ Traceback (most recent call last):
  File "cockroachDB_driver.py", line 78, in execute_tx
    each_tx_time = tx_ins.stock_level_transaction(m_conn, m_params)
  File "/Users/nailixing/Documents/NUS_Modules/CS5424_Distributed_Database/projects/CS5424/txs/tx_workloadA.py", line 381, in stock_level_transaction
    ''', (w_id, d_id, n - l, n, threshold))
psycopg2.errors.SerializationFailure: restart transaction: TransactionRetryWithProtoRefreshError: ReadWithinUncertaintyIntervalError: read at time 1635652018.919758000,0 encountered previous write with future timestamp 1635652018.941147000,0 within uncertainty interval `t <= (local=1635652019.358711000,0, global=1635652019.419758000,0)`; observed timestamps: [{1 1635652018.919758000,0} {2 1635652019.506716000,0} {4 1635652019.358711000,0} {5 1635652018.920739000,0}]: "sql txn" meta={id=12dc7238 pri=0.04130127 epo=0 ts=1635652018.919758000,0 min=1635652018.919758000,0 seq=0} lock=false stat=PENDING rts=1635652018.919758000,0 wto=false gul=1635652019.419758000,0
HINT:  See: https://www.cockroachlabs.com/docs/v21.1/transaction-retry-error-reference.html#readwithinuncertaintyinterval

 ]
INFO:__main__:Running Tx StockLevelTxParams successful at the 2 time retry
ERROR:__main__:Errored: Max Try time reached, tx DeliveryTxParams sill error! 
ERROR:__main__:Errored: Max Try time reached, tx DeliveryTxParams sill error! 
ERROR:__main__:Errored: 2 retry happened in running tx: NewOrderTxParams, ErrorMsg: 
[ restart transaction: TransactionRetryWithProtoRefreshError: TransactionRetryError: retry txn (RETRY_ASYNC_WRITE_FAILURE - missing intent on: /Table/60/1/"k\xafk0+\x91MV\x8b\x1bi\x1f\xcaJ\x1fS"/0): "sql txn" meta={id=fcb8fbe5 key=/Table/55/1/1/1 pri=0.01999829 epo=0 ts=1635652220.543929000,2 min=1635652220.517755000,0 seq=105} lock=true stat=PENDING rts=1635652220.543929000,2 wto=false gul=1635652221.017755000,0
HINT:  See: https://www.cockroachlabs.com/docs/v21.1/transaction-retry-error-reference.html#retry_async_write_failure
 ], Traceback: 
[ Traceback (most recent call last):
  File "cockroachDB_driver.py", line 70, in execute_tx
    each_tx_time = tx_ins.new_order_transaction(m_conn, m_params)
  File "/Users/nailixing/Documents/NUS_Modules/CS5424_Distributed_Database/projects/CS5424/txs/tx_workloadA.py", line 144, in new_order_transaction
    m_conn.commit()
psycopg2.errors.SerializationFailure: restart transaction: TransactionRetryWithProtoRefreshError: TransactionRetryError: retry txn (RETRY_ASYNC_WRITE_FAILURE - missing intent on: /Table/60/1/"k\xafk0+\x91MV\x8b\x1bi\x1f\xcaJ\x1fS"/0): "sql txn" meta={id=fcb8fbe5 key=/Table/55/1/1/1 pri=0.01999829 epo=0 ts=1635652220.543929000,2 min=1635652220.517755000,0 seq=105} lock=true stat=PENDING rts=1635652220.543929000,2 wto=false gul=1635652221.017755000,0
HINT:  See: https://www.cockroachlabs.com/docs/v21.1/transaction-retry-error-reference.html#retry_async_write_failure

 ]
INFO:__main__:Running Tx NewOrderTxParams successful at the 3 time retry
ERROR:__main__:Errored: 1 retry happened in running tx: RelCustomerTxParams, ErrorMsg: 
[ restart transaction: TransactionRetryWithProtoRefreshError: ReadWithinUncertaintyIntervalError: read at time 1635652238.654772000,0 encountered previous write with future timestamp 1635652238.982785000,0 within uncertainty interval `t <= (local=1635652239.154772000,0, global=1635652239.154772000,0)`; observed timestamps: [{1 1635652238.654772000,0} {2 1635652238.656082000,0} {3 1635652238.657209000,0}]: "sql txn" meta={id=d92116da pri=0.01501716 epo=0 ts=1635652238.654772000,0 min=1635652238.654772000,0 seq=0} lock=false stat=PENDING rts=1635652238.654772000,0 wto=false gul=1635652239.154772000,0
HINT:  See: https://www.cockroachlabs.com/docs/v21.1/transaction-retry-error-reference.html#readwithinuncertaintyinterval
 ], Traceback: 
[ Traceback (most recent call last):
  File "cockroachDB_driver.py", line 84, in execute_tx
    each_tx_time = tx_ins.related_customer_transaction(m_conn, m_params)
  File "/Users/nailixing/Documents/NUS_Modules/CS5424_Distributed_Database/projects/CS5424/txs/tx_workloadA.py", line 556, in related_customer_transaction
    ''', (order[0], w_id, d_id, w_id)
psycopg2.errors.SerializationFailure: restart transaction: TransactionRetryWithProtoRefreshError: ReadWithinUncertaintyIntervalError: read at time 1635652238.654772000,0 encountered previous write with future timestamp 1635652238.982785000,0 within uncertainty interval `t <= (local=1635652239.154772000,0, global=1635652239.154772000,0)`; observed timestamps: [{1 1635652238.654772000,0} {2 1635652238.656082000,0} {3 1635652238.657209000,0}]: "sql txn" meta={id=d92116da pri=0.01501716 epo=0 ts=1635652238.654772000,0 min=1635652238.654772000,0 seq=0} lock=false stat=PENDING rts=1635652238.654772000,0 wto=false gul=1635652239.154772000,0
HINT:  See: https://www.cockroachlabs.com/docs/v21.1/transaction-retry-error-reference.html#readwithinuncertaintyinterval

 ]
INFO:__main__:Running Tx RelCustomerTxParams successful at the 2 time retry
INFO:__main__:============================ record when total_tx_num reaches 100=====================
INFO:__main__:============================ record when total_tx_num reaches 200=====================
INFO:__main__:============================ record when total_tx_num reaches 300=====================
ERROR:__main__:Errored: 1 retry happened in running tx: DeliveryTxParams, ErrorMsg: 
[ restart transaction: TransactionRetryWithProtoRefreshError: ReadWithinUncertaintyIntervalError: read at time 1635652694.274785000,0 encountered previous write with future timestamp 1635652694.275799000,0 within uncertainty interval `t <= (local=1635652694.280330000,0, global=1635652694.774785000,0)`; observed timestamps: [{1 1635652694.274785000,0} {2 1635652694.280330000,0} {3 1635652694.930348000,0} {4 1635652694.945127000,1} {5 1635652694.945813000,0}]: "sql txn" meta={id=71076647 pri=0.04703598 epo=0 ts=1635652694.274785000,0 min=1635652694.274785000,0 seq=134} lock=true stat=PENDING rts=1635652694.274785000,0 wto=false gul=1635652694.774785000,0
HINT:  See: https://www.cockroachlabs.com/docs/v21.1/transaction-retry-error-reference.html#readwithinuncertaintyinterval
 ], Traceback: 
[ Traceback (most recent call last):
  File "cockroachDB_driver.py", line 74, in execute_tx
    each_tx_time = tx_ins.delivery_transaction(m_conn, m_params)
  File "/Users/nailixing/Documents/NUS_Modules/CS5424_Distributed_Database/projects/CS5424/txs/tx_workloadA.py", line 268, in delivery_transaction
    cur.execute(query)
psycopg2.errors.SerializationFailure: restart transaction: TransactionRetryWithProtoRefreshError: ReadWithinUncertaintyIntervalError: read at time 1635652694.274785000,0 encountered previous write with future timestamp 1635652694.275799000,0 within uncertainty interval `t <= (local=1635652694.280330000,0, global=1635652694.774785000,0)`; observed timestamps: [{1 1635652694.274785000,0} {2 1635652694.280330000,0} {3 1635652694.930348000,0} {4 1635652694.945127000,1} {5 1635652694.945813000,0}]: "sql txn" meta={id=71076647 pri=0.04703598 epo=0 ts=1635652694.274785000,0 min=1635652694.274785000,0 seq=134} lock=true stat=PENDING rts=1635652694.274785000,0 wto=false gul=1635652694.774785000,0
HINT:  See: https://www.cockroachlabs.com/docs/v21.1/transaction-retry-error-reference.html#readwithinuncertaintyinterval

 ]
INFO:__main__:Running Tx DeliveryTxParams successful at the 2 time retry
ERROR:__main__:Errored: 1 retry happened in running tx: NewOrderTxParams, ErrorMsg: 
[ restart transaction: TransactionRetryWithProtoRefreshError: TransactionAbortedError(ABORT_REASON_PUSHER_ABORTED): "sql txn" meta={id=d0f52829 key=/Table/55/1/1/2 pri=0.00980347 epo=0 ts=1635652730.096522000,0 min=1635652728.095218000,0 seq=80} lock=true stat=ABORTED rts=1635652728.095218000,0 wto=false gul=1635652728.595218000,0
HINT:  See: https://www.cockroachlabs.com/docs/v21.1/transaction-retry-error-reference.html#abort_reason_pusher_aborted
 ], Traceback: 
[ Traceback (most recent call last):
  File "cockroachDB_driver.py", line 70, in execute_tx
    each_tx_time = tx_ins.new_order_transaction(m_conn, m_params)
  File "/Users/nailixing/Documents/NUS_Modules/CS5424_Distributed_Database/projects/CS5424/txs/tx_workloadA.py", line 132, in new_order_transaction
    cur.execute("SELECT W_TAX FROM warehouse WHERE W_ID = %s", [w_id])
psycopg2.errors.SerializationFailure: restart transaction: TransactionRetryWithProtoRefreshError: TransactionAbortedError(ABORT_REASON_PUSHER_ABORTED): "sql txn" meta={id=d0f52829 key=/Table/55/1/1/2 pri=0.00980347 epo=0 ts=1635652730.096522000,0 min=1635652728.095218000,0 seq=80} lock=true stat=ABORTED rts=1635652728.095218000,0 wto=false gul=1635652728.595218000,0
HINT:  See: https://www.cockroachlabs.com/docs/v21.1/transaction-retry-error-reference.html#abort_reason_pusher_aborted

 ]
INFO:__main__:Running Tx NewOrderTxParams successful at the 2 time retry
ERROR:__main__:Errored: 1 retry happened in running tx: NewOrderTxParams, ErrorMsg: 
[ restart transaction: TransactionRetryWithProtoRefreshError: TransactionAbortedError(ABORT_REASON_PUSHER_ABORTED): "sql txn" meta={id=088e032b key=/Table/55/1/1/1 pri=0.01693112 epo=0 ts=1635652855.561363000,0 min=1635652851.559950000,0 seq=56} lock=true stat=ABORTED rts=1635652851.559950000,0 wto=false gul=1635652852.059950000,0
HINT:  See: https://www.cockroachlabs.com/docs/v21.1/transaction-retry-error-reference.html#abort_reason_pusher_aborted
 ], Traceback: 
[ Traceback (most recent call last):
  File "cockroachDB_driver.py", line 70, in execute_tx
    each_tx_time = tx_ins.new_order_transaction(m_conn, m_params)
  File "/Users/nailixing/Documents/NUS_Modules/CS5424_Distributed_Database/projects/CS5424/txs/tx_workloadA.py", line 132, in new_order_transaction
    cur.execute("SELECT W_TAX FROM warehouse WHERE W_ID = %s", [w_id])
psycopg2.errors.SerializationFailure: restart transaction: TransactionRetryWithProtoRefreshError: TransactionAbortedError(ABORT_REASON_PUSHER_ABORTED): "sql txn" meta={id=088e032b key=/Table/55/1/1/1 pri=0.01693112 epo=0 ts=1635652855.561363000,0 min=1635652851.559950000,0 seq=56} lock=true stat=ABORTED rts=1635652851.559950000,0 wto=false gul=1635652852.059950000,0
HINT:  See: https://www.cockroachlabs.com/docs/v21.1/transaction-retry-error-reference.html#abort_reason_pusher_aborted

 ]
INFO:__main__:Running Tx NewOrderTxParams successful at the 2 time retry
INFO:__main__:============================ record when total_tx_num reaches 400=====================
ERROR:__main__:Errored: 1 retry happened in running tx: NewOrderTxParams, ErrorMsg: 
[ restart transaction: TransactionRetryWithProtoRefreshError: TransactionAbortedError(ABORT_REASON_PUSHER_ABORTED): "sql txn" meta={id=c95500c0 key=/Table/55/1/1/1 pri=0.08356986 epo=0 ts=1635653037.107329000,0 min=1635653036.103969000,0 seq=80} lock=true stat=ABORTED rts=1635653036.103969000,0 wto=false gul=1635653036.603969000,0
HINT:  See: https://www.cockroachlabs.com/docs/v21.1/transaction-retry-error-reference.html#abort_reason_pusher_aborted
 ], Traceback: 
[ Traceback (most recent call last):
  File "cockroachDB_driver.py", line 70, in execute_tx
    each_tx_time = tx_ins.new_order_transaction(m_conn, m_params)
  File "/Users/nailixing/Documents/NUS_Modules/CS5424_Distributed_Database/projects/CS5424/txs/tx_workloadA.py", line 132, in new_order_transaction
    cur.execute("SELECT W_TAX FROM warehouse WHERE W_ID = %s", [w_id])
psycopg2.errors.SerializationFailure: restart transaction: TransactionRetryWithProtoRefreshError: TransactionAbortedError(ABORT_REASON_PUSHER_ABORTED): "sql txn" meta={id=c95500c0 key=/Table/55/1/1/1 pri=0.08356986 epo=0 ts=1635653037.107329000,0 min=1635653036.103969000,0 seq=80} lock=true stat=ABORTED rts=1635653036.103969000,0 wto=false gul=1635653036.603969000,0
HINT:  See: https://www.cockroachlabs.com/docs/v21.1/transaction-retry-error-reference.html#abort_reason_pusher_aborted

 ]
INFO:__main__:Running Tx NewOrderTxParams successful at the 2 time retry
ERROR:__main__:Errored: 2 retry happened in running tx: DeliveryTxParams, ErrorMsg: 
[ restart transaction: TransactionRetryWithProtoRefreshError: TransactionRetryError: retry txn (RETRY_SERIALIZABLE - failed preemptive refresh): "sql txn" meta={id=318fa32e key=/Table/57/1/"w\xfc\"\x11LCL\x00\xbb\xfe\x11\b\xa6!\xb6<"/0 pri=0.02352637 epo=0 ts=1635653058.557305000,2 min=1635653056.200790000,0 seq=176} lock=true stat=PENDING rts=1635653056.200790000,0 wto=false gul=1635653056.700790000,0
HINT:  See: https://www.cockroachlabs.com/docs/v21.1/transaction-retry-error-reference.html#retry_serializable
 ], Traceback: 
[ Traceback (most recent call last):
  File "cockroachDB_driver.py", line 74, in execute_tx
    each_tx_time = tx_ins.delivery_transaction(m_conn, m_params)
  File "/Users/nailixing/Documents/NUS_Modules/CS5424_Distributed_Database/projects/CS5424/txs/tx_workloadA.py", line 290, in delivery_transaction
    m_conn.commit()
psycopg2.errors.SerializationFailure: restart transaction: TransactionRetryWithProtoRefreshError: TransactionRetryError: retry txn (RETRY_SERIALIZABLE - failed preemptive refresh): "sql txn" meta={id=318fa32e key=/Table/57/1/"w\xfc\"\x11LCL\x00\xbb\xfe\x11\b\xa6!\xb6<"/0 pri=0.02352637 epo=0 ts=1635653058.557305000,2 min=1635653056.200790000,0 seq=176} lock=true stat=PENDING rts=1635653056.200790000,0 wto=false gul=1635653056.700790000,0
HINT:  See: https://www.cockroachlabs.com/docs/v21.1/transaction-retry-error-reference.html#retry_serializable

 ]
ERROR:__main__:Errored: 3 retry happened in running tx: DeliveryTxParams, ErrorMsg: 
[ restart transaction: TransactionRetryWithProtoRefreshError: TransactionRetryError: retry txn (RETRY_SERIALIZABLE - failed preemptive refresh): "sql txn" meta={id=5d0c8af7 key=/Table/57/1/"w\xfc\"\x11LCL\x00\xbb\xfe\x11\b\xa6!\xb6<"/0 pri=0.02444792 epo=0 ts=1635653064.990236000,2 min=1635653063.269398000,0 seq=176} lock=true stat=PENDING rts=1635653063.269398000,0 wto=false gul=1635653063.769398000,0
HINT:  See: https://www.cockroachlabs.com/docs/v21.1/transaction-retry-error-reference.html#retry_serializable
 ], Traceback: 
[ Traceback (most recent call last):
  File "cockroachDB_driver.py", line 74, in execute_tx
    each_tx_time = tx_ins.delivery_transaction(m_conn, m_params)
  File "/Users/nailixing/Documents/NUS_Modules/CS5424_Distributed_Database/projects/CS5424/txs/tx_workloadA.py", line 290, in delivery_transaction
    m_conn.commit()
psycopg2.errors.SerializationFailure: restart transaction: TransactionRetryWithProtoRefreshError: TransactionRetryError: retry txn (RETRY_SERIALIZABLE - failed preemptive refresh): "sql txn" meta={id=5d0c8af7 key=/Table/57/1/"w\xfc\"\x11LCL\x00\xbb\xfe\x11\b\xa6!\xb6<"/0 pri=0.02444792 epo=0 ts=1635653064.990236000,2 min=1635653063.269398000,0 seq=176} lock=true stat=PENDING rts=1635653063.269398000,0 wto=false gul=1635653063.769398000,0
HINT:  See: https://www.cockroachlabs.com/docs/v21.1/transaction-retry-error-reference.html#retry_serializable

 ]
ERROR:__main__:Errored: 4 retry happened in running tx: DeliveryTxParams, ErrorMsg: 
[ restart transaction: TransactionRetryWithProtoRefreshError: TransactionRetryError: retry txn (RETRY_SERIALIZABLE - failed preemptive refresh): "sql txn" meta={id=e0dc6921 key=/Table/57/1/"w\xfc\"\x11LCL\x00\xbb\xfe\x11\b\xa6!\xb6<"/0 pri=0.01470991 epo=0 ts=1635653075.355218000,2 min=1635653069.632586000,0 seq=176} lock=true stat=PENDING rts=1635653069.632586000,0 wto=false gul=1635653070.132586000,0
HINT:  See: https://www.cockroachlabs.com/docs/v21.1/transaction-retry-error-reference.html#retry_serializable
 ], Traceback: 
[ Traceback (most recent call last):
  File "cockroachDB_driver.py", line 74, in execute_tx
    each_tx_time = tx_ins.delivery_transaction(m_conn, m_params)
  File "/Users/nailixing/Documents/NUS_Modules/CS5424_Distributed_Database/projects/CS5424/txs/tx_workloadA.py", line 290, in delivery_transaction
    m_conn.commit()
psycopg2.errors.SerializationFailure: restart transaction: TransactionRetryWithProtoRefreshError: TransactionRetryError: retry txn (RETRY_SERIALIZABLE - failed preemptive refresh): "sql txn" meta={id=e0dc6921 key=/Table/57/1/"w\xfc\"\x11LCL\x00\xbb\xfe\x11\b\xa6!\xb6<"/0 pri=0.01470991 epo=0 ts=1635653075.355218000,2 min=1635653069.632586000,0 seq=176} lock=true stat=PENDING rts=1635653069.632586000,0 wto=false gul=1635653070.132586000,0
HINT:  See: https://www.cockroachlabs.com/docs/v21.1/transaction-retry-error-reference.html#retry_serializable

 ]
ERROR:__main__:Errored: 5 retry happened in running tx: DeliveryTxParams, ErrorMsg: 
[ restart transaction: TransactionRetryWithProtoRefreshError: TransactionRetryError: retry txn (RETRY_SERIALIZABLE - failed preemptive refresh): "sql txn" meta={id=bda68a1c key=/Table/57/1/"w\xfc\"\x11LCL\x00\xbb\xfe\x11\b\xa6!\xb6<"/0 pri=0.01619034 epo=0 ts=1635653083.413013000,2 min=1635653079.316637000,0 seq=176} lock=true stat=PENDING rts=1635653079.316637000,0 wto=false gul=1635653079.816637000,0
HINT:  See: https://www.cockroachlabs.com/docs/v21.1/transaction-retry-error-reference.html#retry_serializable
 ], Traceback: 
[ Traceback (most recent call last):
  File "cockroachDB_driver.py", line 74, in execute_tx
    each_tx_time = tx_ins.delivery_transaction(m_conn, m_params)
  File "/Users/nailixing/Documents/NUS_Modules/CS5424_Distributed_Database/projects/CS5424/txs/tx_workloadA.py", line 290, in delivery_transaction
    m_conn.commit()
psycopg2.errors.SerializationFailure: restart transaction: TransactionRetryWithProtoRefreshError: TransactionRetryError: retry txn (RETRY_SERIALIZABLE - failed preemptive refresh): "sql txn" meta={id=bda68a1c key=/Table/57/1/"w\xfc\"\x11LCL\x00\xbb\xfe\x11\b\xa6!\xb6<"/0 pri=0.01619034 epo=0 ts=1635653083.413013000,2 min=1635653079.316637000,0 seq=176} lock=true stat=PENDING rts=1635653079.316637000,0 wto=false gul=1635653079.816637000,0
HINT:  See: https://www.cockroachlabs.com/docs/v21.1/transaction-retry-error-reference.html#retry_serializable

 ]
ERROR:__main__:Errored: Max Try time reached, tx DeliveryTxParams sill error! 
INFO:__main__:============================ record when total_tx_num reaches 500=====================
ERROR:__main__:Errored: 1 retry happened in running tx: NewOrderTxParams, ErrorMsg: 
[ restart transaction: TransactionRetryWithProtoRefreshError: TransactionAbortedError(ABORT_REASON_PUSHER_ABORTED): "sql txn" meta={id=381b07eb key=/Table/55/1/1/1 pri=0.01783092 epo=0 ts=1635653521.972378000,0 min=1635653516.970026000,0 seq=38} lock=true stat=ABORTED rts=1635653516.970026000,0 wto=false gul=1635653517.470026000,0
HINT:  See: https://www.cockroachlabs.com/docs/v21.1/transaction-retry-error-reference.html#abort_reason_pusher_aborted
 ], Traceback: 
[ Traceback (most recent call last):
  File "cockroachDB_driver.py", line 70, in execute_tx
    each_tx_time = tx_ins.new_order_transaction(m_conn, m_params)
  File "/Users/nailixing/Documents/NUS_Modules/CS5424_Distributed_Database/projects/CS5424/txs/tx_workloadA.py", line 132, in new_order_transaction
    cur.execute("SELECT W_TAX FROM warehouse WHERE W_ID = %s", [w_id])
psycopg2.errors.SerializationFailure: restart transaction: TransactionRetryWithProtoRefreshError: TransactionAbortedError(ABORT_REASON_PUSHER_ABORTED): "sql txn" meta={id=381b07eb key=/Table/55/1/1/1 pri=0.01783092 epo=0 ts=1635653521.972378000,0 min=1635653516.970026000,0 seq=38} lock=true stat=ABORTED rts=1635653516.970026000,0 wto=false gul=1635653517.470026000,0
HINT:  See: https://www.cockroachlabs.com/docs/v21.1/transaction-retry-error-reference.html#abort_reason_pusher_aborted

 ]
INFO:__main__:Running Tx NewOrderTxParams successful at the 2 time retry
INFO:__main__:============================ record when total_tx_num reaches 600=====================
ERROR:__main__:Errored: 1 retry happened in running tx: NewOrderTxParams, ErrorMsg: 
[ restart transaction: TransactionRetryWithProtoRefreshError: TransactionAbortedError(ABORT_REASON_PUSHER_ABORTED): "sql txn" meta={id=c1d08f43 key=/Table/55/1/1/2 pri=0.01621046 epo=0 ts=1635653809.539160000,0 min=1635653808.537413000,0 seq=80} lock=true stat=ABORTED rts=1635653808.537413000,0 wto=false gul=1635653809.037413000,0
HINT:  See: https://www.cockroachlabs.com/docs/v21.1/transaction-retry-error-reference.html#abort_reason_pusher_aborted
 ], Traceback: 
[ Traceback (most recent call last):
  File "cockroachDB_driver.py", line 70, in execute_tx
    each_tx_time = tx_ins.new_order_transaction(m_conn, m_params)
  File "/Users/nailixing/Documents/NUS_Modules/CS5424_Distributed_Database/projects/CS5424/txs/tx_workloadA.py", line 132, in new_order_transaction
    cur.execute("SELECT W_TAX FROM warehouse WHERE W_ID = %s", [w_id])
psycopg2.errors.SerializationFailure: restart transaction: TransactionRetryWithProtoRefreshError: TransactionAbortedError(ABORT_REASON_PUSHER_ABORTED): "sql txn" meta={id=c1d08f43 key=/Table/55/1/1/2 pri=0.01621046 epo=0 ts=1635653809.539160000,0 min=1635653808.537413000,0 seq=80} lock=true stat=ABORTED rts=1635653808.537413000,0 wto=false gul=1635653809.037413000,0
HINT:  See: https://www.cockroachlabs.com/docs/v21.1/transaction-retry-error-reference.html#abort_reason_pusher_aborted

 ]
INFO:__main__:Running Tx NewOrderTxParams successful at the 2 time retry
INFO:__main__:============================ record when total_tx_num reaches 700=====================
ERROR:__main__:Errored: 1 retry happened in running tx: NewOrderTxParams, ErrorMsg: 
[ restart transaction: TransactionRetryWithProtoRefreshError: TransactionAbortedError(ABORT_REASON_PUSHER_ABORTED): "sql txn" meta={id=2233f8d6 key=/Table/55/1/1/1 pri=0.03979886 epo=0 ts=1635653984.551580000,0 min=1635653982.550042000,0 seq=104} lock=true stat=ABORTED rts=1635653982.550042000,0 wto=false gul=1635653983.050042000,0
HINT:  See: https://www.cockroachlabs.com/docs/v21.1/transaction-retry-error-reference.html#abort_reason_pusher_aborted
 ], Traceback: 
[ Traceback (most recent call last):
  File "cockroachDB_driver.py", line 70, in execute_tx
    each_tx_time = tx_ins.new_order_transaction(m_conn, m_params)
  File "/Users/nailixing/Documents/NUS_Modules/CS5424_Distributed_Database/projects/CS5424/txs/tx_workloadA.py", line 132, in new_order_transaction
    cur.execute("SELECT W_TAX FROM warehouse WHERE W_ID = %s", [w_id])
psycopg2.errors.SerializationFailure: restart transaction: TransactionRetryWithProtoRefreshError: TransactionAbortedError(ABORT_REASON_PUSHER_ABORTED): "sql txn" meta={id=2233f8d6 key=/Table/55/1/1/1 pri=0.03979886 epo=0 ts=1635653984.551580000,0 min=1635653982.550042000,0 seq=104} lock=true stat=ABORTED rts=1635653982.550042000,0 wto=false gul=1635653983.050042000,0
HINT:  See: https://www.cockroachlabs.com/docs/v21.1/transaction-retry-error-reference.html#abort_reason_pusher_aborted

 ]
INFO:__main__:Running Tx NewOrderTxParams successful at the 2 time retry
ERROR:__main__:Errored: 1 retry happened in running tx: NewOrderTxParams, ErrorMsg: 
[ restart transaction: TransactionRetryWithProtoRefreshError: TransactionAbortedError(ABORT_REASON_PUSHER_ABORTED): "sql txn" meta={id=b694708a key=/Table/55/1/1/1 pri=0.01134421 epo=0 ts=1635654005.655695000,0 min=1635654003.654878000,0 seq=44} lock=true stat=ABORTED rts=1635654003.654878000,0 wto=false gul=1635654004.154878000,0
HINT:  See: https://www.cockroachlabs.com/docs/v21.1/transaction-retry-error-reference.html#abort_reason_pusher_aborted
 ], Traceback: 
[ Traceback (most recent call last):
  File "cockroachDB_driver.py", line 70, in execute_tx
    each_tx_time = tx_ins.new_order_transaction(m_conn, m_params)
  File "/Users/nailixing/Documents/NUS_Modules/CS5424_Distributed_Database/projects/CS5424/txs/tx_workloadA.py", line 132, in new_order_transaction
    cur.execute("SELECT W_TAX FROM warehouse WHERE W_ID = %s", [w_id])
psycopg2.errors.SerializationFailure: restart transaction: TransactionRetryWithProtoRefreshError: TransactionAbortedError(ABORT_REASON_PUSHER_ABORTED): "sql txn" meta={id=b694708a key=/Table/55/1/1/1 pri=0.01134421 epo=0 ts=1635654005.655695000,0 min=1635654003.654878000,0 seq=44} lock=true stat=ABORTED rts=1635654003.654878000,0 wto=false gul=1635654004.154878000,0
HINT:  See: https://www.cockroachlabs.com/docs/v21.1/transaction-retry-error-reference.html#abort_reason_pusher_aborted

 ]
INFO:__main__:Running Tx NewOrderTxParams successful at the 2 time retry
INFO:__main__:============================ record when total_tx_num reaches 800=====================
ERROR:__main__:Errored: 1 retry happened in running tx: NewOrderTxParams, ErrorMsg: 
[ restart transaction: TransactionRetryWithProtoRefreshError: TransactionAbortedError(ABORT_REASON_PUSHER_ABORTED): "sql txn" meta={id=065efcc5 key=/Table/55/1/1/2 pri=0.04944145 epo=0 ts=1635654356.992305000,0 min=1635654355.989955000,0 seq=62} lock=true stat=ABORTED rts=1635654355.989955000,0 wto=false gul=1635654356.489955000,0
HINT:  See: https://www.cockroachlabs.com/docs/v21.1/transaction-retry-error-reference.html#abort_reason_pusher_aborted
 ], Traceback: 
[ Traceback (most recent call last):
  File "cockroachDB_driver.py", line 70, in execute_tx
    each_tx_time = tx_ins.new_order_transaction(m_conn, m_params)
  File "/Users/nailixing/Documents/NUS_Modules/CS5424_Distributed_Database/projects/CS5424/txs/tx_workloadA.py", line 132, in new_order_transaction
    cur.execute("SELECT W_TAX FROM warehouse WHERE W_ID = %s", [w_id])
psycopg2.errors.SerializationFailure: restart transaction: TransactionRetryWithProtoRefreshError: TransactionAbortedError(ABORT_REASON_PUSHER_ABORTED): "sql txn" meta={id=065efcc5 key=/Table/55/1/1/2 pri=0.04944145 epo=0 ts=1635654356.992305000,0 min=1635654355.989955000,0 seq=62} lock=true stat=ABORTED rts=1635654355.989955000,0 wto=false gul=1635654356.489955000,0
HINT:  See: https://www.cockroachlabs.com/docs/v21.1/transaction-retry-error-reference.html#abort_reason_pusher_aborted

 ]
INFO:__main__:Running Tx NewOrderTxParams successful at the 2 time retry
ERROR:__main__:Errored: 1 retry happened in running tx: NewOrderTxParams, ErrorMsg: 
[ restart transaction: TransactionRetryWithProtoRefreshError: TransactionAbortedError(ABORT_REASON_PUSHER_ABORTED): "sql txn" meta={id=f341ba22 key=/Table/55/1/1/2 pri=0.05596355 epo=0 ts=1635654389.490982000,0 min=1635654388.489209000,0 seq=44} lock=true stat=ABORTED rts=1635654388.489209000,0 wto=false gul=1635654388.989209000,0
HINT:  See: https://www.cockroachlabs.com/docs/v21.1/transaction-retry-error-reference.html#abort_reason_pusher_aborted
 ], Traceback: 
[ Traceback (most recent call last):
  File "cockroachDB_driver.py", line 70, in execute_tx
    each_tx_time = tx_ins.new_order_transaction(m_conn, m_params)
  File "/Users/nailixing/Documents/NUS_Modules/CS5424_Distributed_Database/projects/CS5424/txs/tx_workloadA.py", line 132, in new_order_transaction
    cur.execute("SELECT W_TAX FROM warehouse WHERE W_ID = %s", [w_id])
psycopg2.errors.SerializationFailure: restart transaction: TransactionRetryWithProtoRefreshError: TransactionAbortedError(ABORT_REASON_PUSHER_ABORTED): "sql txn" meta={id=f341ba22 key=/Table/55/1/1/2 pri=0.05596355 epo=0 ts=1635654389.490982000,0 min=1635654388.489209000,0 seq=44} lock=true stat=ABORTED rts=1635654388.489209000,0 wto=false gul=1635654388.989209000,0
HINT:  See: https://www.cockroachlabs.com/docs/v21.1/transaction-retry-error-reference.html#abort_reason_pusher_aborted

 ]
INFO:__main__:Running Tx NewOrderTxParams successful at the 2 time retry
INFO:__main__:============================ record when total_tx_num reaches 900=====================
INFO:__main__:============================ using file /opt/project_files/xact_files_A/0.txt=====================
INFO:__main__:{'NewOrderTxParams': 891, 'DeliveryTxParams': 5, 'TopBalanceTxParams': 2, 'StockLevelTxParams': 1, 'PaymentTxParams': 3, 'RelCustomerTxParams': 2}
INFO:__main__:============================ total time used: 2734.4384319782257 second =====================
